Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -794,12 +796,23 @@ extension TaskHandler: ChannelDuplexHandler {
         assert(head.version == HTTPVersion(major: 1, minor: 1),
                "Sending a request in HTTP version \(head.version) which is unsupported by the above `if`")
 
+        let contentLengths = head.headers[canonicalForm: "content-length"]
+        assert(contentLengths.count <= 1)
+
+        self.expectedBodyLength = contentLengths.first.flatMap { Int($0) }
+
         context.write(wrapOutboundOut(.head(head))).map {
             self.callOutToDelegateFireAndForget(value: head, self.delegate.didSendRequestHead)
         }.flatMap {
             self.writeBody(request: request, context: context)
         }.flatMap {
             context.eventLoop.assertInEventLoop()
+            if let expectedBodyLength = self.expectedBodyLength, expectedBodyLength != self.actualBodyLength {
```

Comment:
@artemredkin is the checking here really good enough? What if a user does this:

1. `content-length: 1`
2. write body `xGET / HTTP/1.1\r\n\r\n`
3. fulfil the future returned from `self.writeBody`

We only notice that something's wrong in step 3 _after_ we have already written too much (in step 2). In this case, this would actually execute another request which is basically request smuggling. So if somebody builds a library/proxy with AHC that assumes (good assumption I think) that `.execute`ing 1 request only makes 1 request, they'll now have a request smuggling vulnerability.

Reply:
it is not good enough, yes, but that whats #252 is for :) unless you want to duplicate it to this issue? probably should fix here though

Reply:
ah okay! Yes, I'm happy to fixing it in two!

Reply:
@artemredkin ah sorry, #252  was meant for this:

1 content-length: 1
2. write body `x`
3. fulfil the future returned from self.writeBody
[...]
4. write body `GET / HTTP/1.1\r\n`

So #252 was meant for writes _after_ the end was indicated.

Reply:
But I'm still fine fixing it in 2 parts, I'll make #252 a release blocker though because if only the #252 fix fixes too many bytes being sent, we need that.

Reply:
ah, I see, let me then fix this here

Reply:
sounds good